### DISCUSSION TITLE
v3.1.5 - Minor improvements

### DISCUSSION THREAD

**Discussion by alexdelprete** [alexdelprete/ha-abb-powerone-pvi-sunspec#85] [Announcements]
- New model added: UNO-DM-5.0-TL-PLUS
- Finalized check_port()
- Removed unused imports

<hr /><em>This discussion was created from the release <a href='https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/releases/tag/v3.1.5'>v3.1.5 - Minor improvements</a>.</em>

**Comment by lucasdaly90** [comment]
Hullo!

My inverter is showing the correct model now. Thank you.

I've noticed occasionally the inverter will report UNAVAILABLE on all sensors.

The Logs show the following: Any ideas? Both HA and the Inverter Ping with no latency difference during these periods. Would I be better to increase the time between the polling intervals? I'm on 45s presently.

************************
This error originated from a custom integration.

Logger: custom_components.abb_powerone_pvi_sunspec
Source: helpers/update_coordinator.py:233
Integration: ABB/Power-One/FIMER PVI SunSpec (documentation, issues)
First occurred: 9:10:28 AM (6 occurrences)
Last logged: 9:58:12 AM

Error fetching abb_powerone_pvi_sunspec data:
************************

**Reply by lucasdaly90** [reply]
Yes, that seems to be ok, but I get the Unavailable issues whilst the sun is up - EG, i'm receiving ~4000w at present, and it went Unavailable for a cycle.

**Reply by alexdelprete** [reply]
Sorry, I thought the inverter was offline, but I see it's 9am over there.

No, it's not normal. Turn on debug logging on the integration, wait 2-3 minutes, then check the full HA logs, search for "abb" so you see only linea related to the integration. Post the log here.

![image](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/7027842/dcb5ae66-f36a-4f4b-a380-805e1c58a2d2)

**Comment by lucasdaly90** [comment]
[home-assistant_abb_powerone_pvi_sunspec_2023-10-01T00-23-18.897Z.log](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/files/12775771/home-assistant_abb_powerone_pvi_sunspec_2023-10-01T00-23-18.897Z.log)

Please see attached.

**Comment by lucasdaly90** [comment]
[home-assistant_2023-10-01T00-26-52.783Z.log](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/files/12775775/home-assistant_2023-10-01T00-26-52.783Z.log)

Actually, this log set may be better.

**Comment by alexdelprete** [comment]
As you can see from the below snippet (see the check_port lines), port 502 on that IP was not available at socket level: I do a check port before connecting, and this means there's something not right at network level. How's the wifi signal of the VSN card? How's the RPi connected, Wifi or ETH?

```
2023-10-01 10:18:24.168 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] ABB SunSpec Update data coordinator update
2023-10-01 10:18:24.169 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] Hub connect to IP: 192.168.2.83 port: 502 slave id: 2 timeout: 44
2023-10-01 10:18:24.169 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] Check_Port: opening socket on 192.168.2.83:502 with a 3.0s timeout.
2023-10-01 10:18:27.174 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] Check_Port (ERROR): port not available on 192.168.2.83:502 - error: 11
2023-10-01 10:18:27.176 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] Inverter not ready for Modbus TCP connection
2023-10-01 10:18:27.176 DEBUG (MainThread) [custom_components.abb_powerone_pvi_sunspec] Async Update Data error: Inverter not active on 192.168.2.83:502
2023-10-01 10:18:27.177 ERROR (MainThread) [custom_components.abb_powerone_pvi_sunspec] Error fetching abb_powerone_pvi_sunspec data:
```

**Comment by alexdelprete** [comment]
Also, this is not a very nice error: RPi already give a lot of weird issues, if you don't power it correctly, it's even worse.

```
[homeassistant.components.rpi_power.binary_sensor] Under-voltage was detected. Consider getting a uninterruptible power supply for your Raspberry Pi.
```

**Comment by lucasdaly90** [comment]
Signal strenth looks ok. UniFi is reporting no drops, and the inverter's interface reports Strong.
I've got a new legit RPI PSU being recieved today. I saw those popping up in the last week too.


<img width="346" alt="Screenshot 2023-10-01 at 10 34 03 am" src="https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/146505997/48acea49-67f0-41e2-9a71-acc3357ec3af">
<img width="376" alt="Screenshot 2023-10-01 at 10 31 49 am" src="https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/146505997/c0b570f6-b190-449a-a1aa-994f3074ea46">

**Reply by alexdelprete** [reply]
Is that the VSN stats from unifi or RPi? How's RPi connected?

check_port() does a socket level connect on port 502 of the VSN card. If it returns an error, it means it can't connect.

I see it happens frequently...you need to dig deeper to understand what's going on...but clearly there's a communication issue.

**Reply by alexdelprete** [reply]
I see this, what does it mean? Poor is not a good indicator...

![image](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/7027842/454203e8-2e35-4b5a-8d2c-07aa66436e70)

**Reply by lucasdaly90** [reply]
I think I'll look into the Ethernet module, they seem cheap and that will take any RF issues out of the equation.

Pi is Ethernet, Inverter is Wifi - the AP is about 4m away in direct line of sight.

**Reply by alexdelprete** [reply]
The undervoltage is an issue too, RPi is very power sensitive. Fix all the issues and you won't have the connection problems.

**Reply by lucasdaly90** [reply]
I'm somewhat ressurecting this thread. I'm constantly getting unavailable for requests from this inverter. I've replaced PSU etc, and that seems to be rock solid now.

[home-assistant_2023-11-28T23-57-29.026Z.log](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/files/13494578/home-assistant_2023-11-28T23-57-29.026Z.log)

This error originated from a custom integration.

Logger: custom_components.abb_powerone_pvi_sunspec
Source: helpers/update_coordinator.py:322
Integration: ABB/Power-One/FIMER PVI SunSpec (documentation, issues)
First occurred: 9:53:38 AM (1 occurrences)
Last logged: 9:53:38 AM

Error fetching abb_powerone_pvi_sunspec data:

**Reply by alexdelprete** [reply]
the log doesn't say anything, enable debug in the integration and restart.

![image](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/7027842/7db3098f-c5da-4af6-a5ca-2e78e186365f)

once you have the debug log I can see what's happening.
